frontend/app/routes/$lang+/_protected+/letters+/index.tsx

```diff
@@ -100,6 +100,7 @@ export default function LettersIndex() {
       </div>
       <ul className="divide-y border-y">
         {letters.map((letter) => {
+          if (letter.name === undefined || letter.name === '') return;
           const letterType = letterTypes.find(({ id }) => id === letter.name);
           const translatedLetterName = letterType ? getNameByLanguage(i18n.language, letterType) : letter.name;
           const letterName = translatedLetterName ?? letter.name;
```

Comment:
With this newly introduced code, if a `letter.name` is found but does not exist in `letterTypes`, we are going to default to displaying the `letter.name` (ie. the ID of the letter type). 

However, the user story indicates if the `letter.name` is not in the Power Platform mock (ie. the value of `letterTypes`), do not display the letter.

I think we should be adding an `if (!letterType) { return <></> }` check instead of what is introduced in this PR.

Reply:
One could make a case for handling some of this mapping inside of the service module. Why not just return a collection of letters with the letter type already pre-mapped. If no type exists, map it to `undefined` and then filter that out during rendering with a `.filter(letter.name && letter.type)` sort of thing.

Reply:
Let me know if you prefer to map this in the service instead.